app/scripts/inpage.js

```diff
@@ -33,13 +33,13 @@ cleanContextForImports();
 /* eslint-disable import/first */
```

Comment:
Interesting I hadn't thought of introducing this via the inpage provider. I had thought more of enabling connections but not actually reconfiguring our inpage to use this connection 🤔

Reply:
To be clear here, i don't intend to merge the inpage changes as part of this PR.

They're only here since I needed them while building out the background js changes. I do think our first `@metamask/providers` release that includes caip enveloping support will including some of these inpage changes